src/osd/PG.cc

```diff
@@ -7724,6 +7733,11 @@ PG::RecoveryState::GetMissing::GetMissing(my_context ctx)
     // We pull the log from the peer's last_epoch_started to ensure we
     // get enough log to detect divergent updates.
     eversion_t since(pi.last_epoch_started, 0);
+    //We can ensure the peer isn't divergent in the following case
```

Comment:
I'm really not convinced that this is correct.  Please add a more detailed explanation of why we can be sure.